test/runnable/test_cdcmp.d

```diff
@@ -1,7 +1,7 @@
 // REQUIRED_ARGS: -O -fPIC
 // PERMUTE_ARGS:
 // only testing on SYSV-ABI, but backend code is identical across platforms
-// DISABLED: win32 win64 osx linux32 freebsd32
+// DISABLED: win32 win64 osx linux32 freebsd32 linux64 freebsd64
```

Comment:
What is the point of this test if it's now disabled on all platforms dmd supports?

Reply:
Because it's a highly annoying test.

1. It breaks every time I try to adjust the code generator.
2. Compiling the updating version of it is a nuisance.
3. Compiling the updating version requires the very latest Phobos to be installed and working.
4. Meanwhile, while trying to get (3) to work, the autotester won't test the rest of the compiler.
5. The compiler test suite shouldn't rely on Phobos, because that makes bootstrapping the compiler hard. Let alone the very latest Phobos (it requires std.typecons.EnumMembers).

I'll deal with it dead last.

Reply:
> the very latest Phobos to be installed and working.

No that test is almost two years old, so the update script can't rely on "the very latest Phobos".

> The compiler test suite shouldn't rely on Phobos, because that makes bootstrapping the compiler hard

You do realize that the very script that you use to b execute the tests has been written in D? (d_do_tests)
Also, this test only requires Phobos for updating itself - not running.

Reply:
DMD 2.060 (from 2012) has a bug fix for EnumMembers: 

https://dlang.org/changelog/2.060.html

I don't have time to look more into the past and you probably still need a recent version of DMD/Phobos, because for example everything before 2.072 was compiled without PIC meaning that you can't run it on any modern Linux distribution. Also our bootstrapping version of DMD is 2.079. That's what all CIs use too. We're not going to cater for anything below 2.079.

Reply:
Basically all of Phobos has to work perfectly for this test to even compile, let alone run. It's just completely wrong for a back end test.

Reply:
@wilzbach that dependency seems onerous.

Reply:
> Basically all of Phobos has to work perfectly for this test to even compile, let alone run. 

__NOPE__: it requires no Phobos at all. Note the `version(update)`:

https://github.com/dlang/dmd/blob/ecbb547a826b4c198de78a91f002d94819d72b9d/test/runnable/test_cdcmp.d#L26-L28

It's only required for the automated update script.

Reply:
The update version has to be run with the compiler under test, and it relies on Phobos.

Reply:
Nope, you set the `DMD` under test via an environment (as the update script header explains) and can thus continue to use the host compiler for compiling the update script.